stubs/SQLAlchemy/sqlalchemy/engine/create.pyi

```diff
@@ -1,2 +1,33 @@
-def create_engine(url, **kwargs): ...
-def engine_from_config(configuration, prefix: str = ..., **kwargs): ...
+from collections.abc import Mapping
+from typing import Any, overload
+from typing_extensions import Literal
+
+from ..future.engine import Engine as FutureEngine
+from .base import Engine
+from .mock import MockConnection
+from .url import URL
+
+# Further kwargs are forwarded to the engine, dialect, or pool.
+@overload
+def create_engine(url: URL | str, *, strategy: Literal["mock"], **kwargs) -> MockConnection: ...  # type: ignore[misc]
```

Comment:
Why does this need a type ignore?

Reply:
mypy: "overlap with incompatible return types"